File: 백준/Silver/10773. 제로/제로.py
# 10773. 제로
# 0을 외치면 가장 최근에 적은 수 지우기
# 정수 K : 테스트 개수
import sys
stack = []
K = int(sys.stdin.readline())
for i in range(K):
    result = int(sys.stdin.readline())
    if result != 0:
        stack.append(result)
    elif result == 0:
        stack.pop() # stack.pop(-1)와 같음

sum = 0
for i in stack:
    sum += i

print(sum)

File: 백준/Silver/10773. 제로/제로.java
import java.io.*;
import java.util.LinkedList;
import java.util.Scanner;
import java.util.Stack;

public class Main {
    static int sum;
    public static void main(String[] args) throws IOException {
        Stack<Integer> stack = new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int k = Integer.parseInt(br.readLine());

        for (int i = 0;  i < k; i++){
            int a = Integer.parseInt(br.readLine());
            if (a == 0){
                stack.pop();
            }
            else {
                stack.push(a);
            }
        }
        for (Integer i : stack) {
            sum += i;
        }
        bw.write(sum + "");
        bw.flush();
        bw.close();
    }
}
